### PR TITLE
Added support for git shortcuts + README

### PR DESCRIPTION
**Pull request by IvanDelsinne**
This PR implements path detection support for git short URLs.

See #19

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -47,7 +48,9 @@ pub fn create_directory(path: &Path) -> Result<()> {
 /// Is it a remote or a local thing
 pub fn get_source(input: &str) -> Source {
     // Should be a Regex once we add hg or other stuff
-    if input.starts_with("git@") || input.starts_with("http://") || input.starts_with("https://") {
+    let git_shortcut_re = Regex::new(r"^\w+:").unwrap();
```
[the remark this answers is not in the document]
Comment (IvanDelsinne) [reply src/utils.rs]: If the only check to determine if it's a Git HTTP/SSH URL is "is there any `:`" (as one shouldn't have any `:` in a folder name), I can even kick out the regex and just do some "does contains"

File: src/utils.rs [hunk 2]
```diff
@@ -47,10 +46,9 @@ pub fn create_directory(path: &Path) -> Result<()> {
 
 /// Is it a remote or a local thing
 pub fn get_source(input: &str) -> Source {
-    // Should be a Regex once we add hg or other stuff
-    let git_shortcut_re = Regex::new(r"^\w+:").unwrap();
-
-    if input.starts_with("git@") || input.starts_with("http://") || input.starts_with("https://") || git_shortcut_re.is_match(input) {
+    // A local path shouldn't be able to contain a : and every Git URL format contains this character,
```
Comment (Keats) [inline src/utils.rs]: A bit late but I just realised `:` is a completely file/path valid character on Linux and OSX but not Windows. I think we will have to check if the path exists for maximal compatibility in the end :/

Comment (IvanDelsinne) [reply src/utils.rs]: I thought about it today while at work too, but that'd require some fair bit of change.

Would it be better to

- merge this PR to "support" the git short tags, and trust users to not use `:` in their file path for now, then work on another PR to re-work the template check flow
- directly re-work the template check flow before merge

I don't know which one to decide, while I think that the first one would be the preferred to start from the cleaned input checking code.

Comment (Keats) [reply src/utils.rs]: Nevermind, this heuristic of checking for `:` doesn't actually work: a path copy pasted on Windows will look like `C:\Program Files\etc` which would be classified as git.

I believe the only way is to load the input into a `Path` and check if it exists

**Comment by IvanDelsinne** [comment]
I don't really know how to make the tests for the full path checking logic. Could you handle that ?

**Pull request merged**